src/wallet/rpc/spend.cpp

```diff
@@ -1159,6 +1160,267 @@ RPCHelpMan send()
     };
 }
 
+RPCHelpMan sweep()
+{
+    return RPCHelpMan{"sweep",
+        "\nEXPERIMENTAL warning: this call may be changed in future releases.\n"
+        "\nSpend all confirmed UTXOs in the wallet to one or more recipients.\n"
+        "Unconfirmed inbound UTXOs and locked UTXOs will not be spent. Sweep will respect the wallet flag for avoid_reuse.\n",
+        {
+            {"receivers", RPCArg::Type::ARR, RPCArg::Optional::NO, "The destinations of the sweep, each address may only appear once.\n"
+                "Optionally some receivers can be specified with an amount, but at least one address must appear without a specified amount.\n",
+                {
+                    {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "A bitcoin address which receives an equal share of the unspecified amount."},
+                    {"", RPCArg::Type::OBJ_USER_KEYS, RPCArg::Optional::OMITTED, "",
+                        {
+                            {"address", RPCArg::Type::AMOUNT, RPCArg::Optional::NO, "A key-value pair. The key (string) is the bitcoin address, the value (float or string) is the amount in " + CURRENCY_UNIT + ""},
+                        },
+                    },
+                },
+            },
+            {"conf_target", RPCArg::Type::NUM, RPCArg::DefaultHint{"wallet -txconfirmtarget"}, "Confirmation target in blocks"},
+            {"estimate_mode", RPCArg::Type::STR, RPCArg::Default{"unset"}, std::string() + "The fee estimate mode, must be one of (case insensitive):\n"
+                        "       \"" + FeeModes("\"\n\"") + "\""},
+            {"fee_rate", RPCArg::Type::AMOUNT, RPCArg::DefaultHint{"not set, fall back to wallet fee estimation"}, "Specify a fee rate in " + CURRENCY_ATOM + "/vB."},
+            {"options", RPCArg::Type::OBJ, RPCArg::Optional::OMITTED_NAMED_ARG, "",
+                {
+                    {"add_to_wallet", RPCArg::Type::BOOL, RPCArg::Default{true}, "When false, returns a serialized transaction which will not be added to the wallet or broadcast"},
+                    {"fee_rate", RPCArg::Type::AMOUNT, RPCArg::DefaultHint{"not set, fall back to wallet fee estimation"}, "Specify a fee rate in " + CURRENCY_ATOM + "/vB."},
```

Comment:
The fee_rate is defined twice:
1. As an argument
2. As an option object

I presume this is an intentional design choice. But I can't seem to grasp the reason behind doing so.

Reply:
> This is consistent with how all other transaction creating RPCs permit setting the `fee_rate`.

I checked with other RPCs where `"fee_rate", RPCArg::Type::AMOUNT,...` is used. It seems like the situation you are describing is true only for the `send` RPC (see https://github.com/bitcoin/bitcoin/blob/master/src/wallet/rpc/spend.cpp#L1050-L1062) and isn't a norm per se.

Other RPCs that uses fee_rate but doesn't define it twice are:
- sendtoaddress
- sendmany
- fundrawtransaction
- bumpfee_helper, and,
- walletcreatefundedpsbt

Reply:
Okay, I'll take another look at that. Thank you.

Reply:
After talking to a few people, my sentiment is that this is the expected behavior. In the minimal use case, the most important arguments are available as positional, making it easy to use. When people start using options, they may want to define everything as options, though, for consistency and readability.